sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -559,15 +559,18 @@ def discard(self, instruction_id, exception):
     """
     Marks the instruction id as failed shutting down the ``BundleProcessor``.
     """
+    processor = None
     with self._lock:
       self.failed_instruction_ids[instruction_id] = exception
       while len(self.failed_instruction_ids) > MAX_FAILED_INSTRUCTIONS:
         self.failed_instruction_ids.popitem(last=False)
-      processor = self.active_bundle_processors[instruction_id][1]
-      del self.active_bundle_processors[instruction_id]
+      if instruction_id in self.active_bundle_processors:
+        processor = self.active_bundle_processors.pop(instruction_id)[1]
```

Comment:
To be more future proof here it seems like you should change the local processor variable to the full list, and then iterate over it to shutdown below.

Alternatively, is there a reason the value in the dict is a list instead of single processor and could that be changed?